File: TowerDefence.spritebuilder/Unit/Solder/SolderFabric.swift
//
//  SolderFabric.swift
//  TowerDefence
//
//  Created by koveres on 07.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation



public class SolderFabric: StackObjectAbstractFabric {
    
    public override  init(intlist:[CInt] )
    {
        super.init(intlist: intlist);
        
    }
    
    public func getUnit( type:SolderType,   startPoint1:CGPoint) -> (Solder)
    {
        var newobj:Solder;
        startPoint = startPoint1;
        newobj =    getObject(type.rawValue) as Solder ;
        return newobj; 
    }
    
    
    var  startPoint:CGPoint?;
   
    override func  getNewObject(  idTypeObject:CInt, grObj:AnyObject) -> (StackObject) {
        var type:SolderType?;
        type = SolderType(rawValue: idTypeObject);
        if(type != nil)
        {
            switch type!
            {
        case SolderType.Solder1:
            return   Solder1(p1: startPoint!,gobj: grObj);
        case SolderType.Solder2:
            return   Solder2(p1: startPoint!,gobj: grObj);
        case SolderType.Solder3:
            return   Solder3(p1: startPoint!,gobj: grObj);
            }
        }
        return   Solder1(p1: startPoint!,gobj: grObj);
    }
}

File: test.spritebuilder/Source/TDEngine/Unit/Solder/SolderFabric.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Unit.Solder;

import Unit.Solder.Solder;
import Utils.Point2D;
import Utils.StackObject;
import Utils.StackObjectAbstractFabric;
import internalObject.SolderInternal.SolderType;
 
 
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author ver
 */
public class SolderFabric extends StackObjectAbstractFabric {
   
    public SolderFabric(List<Integer> intlist)
    {
        super(intlist);
        
    }
    public Solder getUnit(SolderType type,Point2D  startPoint) throws Exception
    {
        Solder newobj;
        this.startPoint = startPoint;
        newobj = (Solder) this.getObject( type.ordinal() );
       newobj.init(startPoint);
        return newobj;
      
    }

    
    Point2D  startPoint;
    @Override
    protected StackObject getNewObject(Integer idObject) {
        SolderType type;
       type = SolderType.values()[idObject];
        switch(type)
        {
            case Solder1:
                return new Solder1(startPoint);
            case Solder2:
                return new Solder2(startPoint);
            case Solder3:
                return new Solder3(startPoint);
        }
        return new Solder1(startPoint);
    }
}
